src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -102,22 +106,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   }
 
   private String getLink(String id, String query) {
-    String cse_request = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
-    HttpRequest linkRequest =
-        HttpRequest.newBuilder()
-            .GET()
-            .uri(URI.create(cse_request))
-            .setHeader("User-Agent", "Java 11 HttpClient Bot")
-            .build();
-    HttpResponse<String> linkResponse;
-    try {
-      linkResponse = httpClient.send(linkRequest, HttpResponse.BodyHandlers.ofString());
-    } catch (IOException | InterruptedException ex) {
+    String url = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      try (CloseableHttpResponse response = httpClient.execute(new HttpGet(url))) {
+
+        if (response.getStatusLine().getStatusCode() != 200) {
+          return null;
+        }
+        HttpEntity entity = response.getEntity();
+        if (entity == null) {
+          return null;
+        }
+        JsonParser parser = new JsonParser();
+        JsonObject obj;
+        try {
+          obj = parser.parse(new InputStreamReader(entity.getContent())).getAsJsonObject();
+        } catch (JsonIOException
+            | JsonSyntaxException
+            | UnsupportedOperationException
+            | IOException e) {
+          return null;
+        }
+        if (!obj.isJsonNull()
+            && !obj.get(CSE_ITEMS).getAsJsonArray().isJsonNull()
+            && obj.get(CSE_ITEMS).getAsJsonArray().size() > 0) {
+          JsonObject item = obj.get(CSE_ITEMS).getAsJsonArray().get(0).getAsJsonObject();
+          if (!item.isJsonNull() && !item.get(CSE_LINK).isJsonNull()) {
+            return item.get(CSE_LINK).getAsString();
+          }
+        }
+      }
+    } catch (IOException e) {
```

Comment:
This catch corresponds to the outer try which can't throw an exception (it just creates a HttpClient). This should correspond to the try that executes the HTTP GET.

Reply:
The outer try can throw an exception due to the try-with structure apparently. Because `Unhandled exception type IOException thrown by automatic close() invocation on httpClientJava(16778098)`

But yes, will add additional cath for the nested try.